netlink/netlink_linux.go

```diff
@@ -1253,25 +1252,33 @@ func SetMacAddress(name, addr string) error {
 }
 
 func SetHairpinMode(iface *net.Interface, enabled bool) error {
-	sysPath := filepath.Join("/sys/class/net", iface.Name, "brport/hairpin_mode")
-
-	sysFile, err := os.OpenFile(sysPath, os.O_WRONLY, 0)
+	s, err := getNetlinkSocket()
 	if err != nil {
 		return err
 	}
-	defer sysFile.Close()
+	defer s.Close()
+	req := newNetlinkRequest(syscall.RTM_SETLINK, syscall.NLM_F_ACK)
 
-	var writeVal []byte
+	msg := newIfInfomsg(syscall.AF_BRIDGE)
+	msg.Type = syscall.RTM_SETLINK
+	msg.Flags = syscall.NLM_F_REQUEST
+	msg.Index = int32(iface.Index)
+	msg.Change = DEFAULT_CHANGE
+	req.AddData(msg)
+
+	mode := []byte{0}
 	if enabled {
-		writeVal = []byte("1")
-	} else {
-		writeVal = []byte("0")
+		mode[0] = byte(1)
 	}
-	if _, err := sysFile.Write(writeVal); err != nil {
+
+	br := newRtAttr(syscall.IFLA_PROTINFO|syscall.NLA_F_NESTED, nil)
+	newRtAttrChild(br, 4, mode)
```

Comment:
The 4 here is IFLA_BRPORT_MODE, right? If so, could we define a constant for it?